bundles/binding/org.openhab.binding.ecotouch/src/main/java/org/openhab/binding/ecotouch/internal/EcoTouchBinding.java

```diff
@@ -124,20 +126,48 @@ protected void execute() {
 						continue;
 					}
 					int heatpumpValue = rawvalues.get(item.getTagName());
+					State value;
 					if (item.getType() == EcoTouchTags.Type.Analog) {
 						// analog value encoded as a scaled integer
 						BigDecimal decimal = new BigDecimal(heatpumpValue)
 								.divide(new BigDecimal(10));
-						handleEventType(new DecimalType(decimal), item);
+						value = new DecimalType(decimal);
 					} else if (item.getType() == EcoTouchTags.Type.Word) {
 						// integer
-						handleEventType(new DecimalType(heatpumpValue), item);
+						if (item.getItemClass() == NumberItem.class)
+							value = new DecimalType(heatpumpValue);
+						else {
+							// assume SwitchItem
+							if (heatpumpValue == 0)
+								value = OnOffType.OFF;
+							else
+								value = OnOffType.ON;
+						}
 					} else {
 						// bit field
 						heatpumpValue >>= item.getBitNum();
 						heatpumpValue &= 1;
-						handleEventType(new DecimalType(heatpumpValue), item);
+						if (item.getItemClass() == NumberItem.class)
```

Comment:
although "==" seems to be valid when comparing classes i would prefer "NumberItem.class.equals(item.getItemClass())"